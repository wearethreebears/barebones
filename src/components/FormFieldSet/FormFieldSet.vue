<template>
  <fieldset :data-component="$options.name" :class="part['component']">
    <legend :class="part['legend']">{{ props.label }}</legend>
    <div :class="part['fields']">
      <slot />
    </div>
  </fieldset>
</template>

<script lang="ts">
export default { name: "BonesFormFieldSet" };
</script>

<script setup lang="ts">
import type { PropType } from "vue";
import useComponent from "@barebones/composables/useComponent";

import { classProps } from "@barebones-local/FormFieldSet/FormFieldSet.props";
import FormFieldSetLocal from "@barebones-local/FormFieldSet/FormFieldSet.parts";

const { parts } = FormFieldSetLocal();

const props = defineProps({
  ...classProps,
  label: {
    required: true,
    type: String as PropType<string>,
  },
});

const { part } = useComponent(parts, props, classProps);
</script>
