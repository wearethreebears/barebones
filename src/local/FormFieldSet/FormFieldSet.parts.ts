import type { Parts } from "@barebones/types/component";
import type { FormFieldSet } from "@barebones/types/components/FormFieldSet";

export default function FormFieldSetLocal(): FormFieldSet {
  const parts: Parts = {
    component: {},
  };

  return {
    parts,
  };
}
