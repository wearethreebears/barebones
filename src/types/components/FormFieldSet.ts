import type { Parts } from "@barebones/types/component";

export interface FormFieldSet {
  parts: Parts;
}
