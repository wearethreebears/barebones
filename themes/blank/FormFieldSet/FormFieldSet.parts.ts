import type { Parts } from "@barebones/types/component";
import type { FormFieldSet } from "@barebones/types/components/FormFieldSet";

// Base styles

export default function FormFieldSetLocal(): FormFieldSet {
  const parts: Parts = {
    component: {},
    label: {},
    fieldsWrapper: {},
  };

  return {
    parts,
  };
}
