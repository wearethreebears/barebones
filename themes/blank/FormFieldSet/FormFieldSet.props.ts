import type { PropType } from "vue";
import type { ClassProps } from "@barebones/types/prop";

export const classProps: ClassProps = {
  style: {
    type: String as PropType<string>,
    default: "default",
  },
};
