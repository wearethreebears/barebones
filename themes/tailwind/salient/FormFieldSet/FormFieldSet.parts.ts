import type { Parts } from "@barebones/types/component";
import type { FormFieldSet } from "@barebones/types/components/FormFieldSet";

// Base styles

export default function FormFieldSetLocal(): FormFieldSet {
  const parts: Parts = {
    component: {
      style: {
        default: ["p-4 border border-gray-200 mb-4"],
      },
    },
    label: {
      style: {
        default: ["text-sky-900 py-2 px-4 border border-gray-200 rounded"],
      },
    },
    fieldsWrapper: {
      default: {
        style: ["flex"],
      },
    },
  };

  return {
    parts,
  };
}
