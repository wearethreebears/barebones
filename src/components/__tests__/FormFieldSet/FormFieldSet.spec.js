import { describe, it, expect, beforeEach } from "vitest";

import { mount } from "@vue/test-utils";
import { allComponentPartClassesToDisplayCorrectly } from "@barebones/components/__tests__/helpers";
import FormFieldSet from "@barebones/components/FormFieldSet/FormFieldSet.vue";

import FormFieldSetLocal from "@barebones-local/FormFieldSet/FormFieldSet.parts";
const { parts } = FormFieldSetLocal();


const defaultSlot = "Hello world";
const props = {
  label: "Movies"
}

allComponentPartClassesToDisplayCorrectly(FormFieldSet, parts);

describe("FormFieldSet", () => {
  let wrapper;
  beforeEach(async () => {
    wrapper = mount(FormFieldSet, {
      props: {
        label: props.label
      },
      slots: {
        default: defaultSlot
      }
    });
  })

  it("renders properly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("displays the correct slot", () => {
    expect(wrapper.find('[ data-part="label"]').html()).contains(props.label)
  });

  it("displays the correct slot", () => {
    expect(wrapper.find('[ data-part="fieldsWrapper"]').html()).contains(defaultSlot)
  });

});
