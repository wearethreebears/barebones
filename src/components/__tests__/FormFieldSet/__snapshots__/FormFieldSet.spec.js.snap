// Vitest Snapshot v1

exports[`FormFieldSet > renders properly 1`] = `
VueWrapper {
  "__app": {
    "_component": {
      "name": "VTU_ROOT",
      "render": [Function],
    },
    "_container": <div
      data-v-app=""
    >
      <fieldset
        class="p-4 border border-gray-200 mb-4"
        data-component="BonesFormFieldSet"
        data-part="component"
      >
        <legend
          class="text-sky-900 py-2 px-4 border border-gray-200 rounded"
          data-part="label"
        >
          Movies
        </legend>
        <div
          class="flex"
          data-part="fieldsWrapper"
        >
          
          Hello world
          
        </div>
      </fieldset>
    </div>,
    "_context": {
      "app": [Circular],
      "components": {
        "transition": {
          "name": "transition",
        },
        "transition-group": {
          "name": "transition-group",
        },
      },
      "config": {
        "compilerOptions": {},
        "errorHandler": undefined,
        "globalProperties": {},
        "isNativeTag": [Function],
        "optionMergeStrategies": {},
        "performance": false,
        "warnHandler": undefined,
      },
      "directives": {},
      "emitsCache": WeakMap {},
      "mixins": [
        {
          "beforeCreate": [Function],
        },
      ],
      "optionsCache": WeakMap {},
      "propsCache": WeakMap {},
      "provides": {},
      "reload": [Function],
    },
    "_instance": {
      "a": null,
      "accessCache": {},
      "appContext": {
        "app": [Circular],
        "components": {
          "transition": {
            "name": "transition",
          },
          "transition-group": {
            "name": "transition-group",
          },
        },
        "config": {
          "compilerOptions": {},
          "errorHandler": undefined,
          "globalProperties": {},
          "isNativeTag": [Function],
          "optionMergeStrategies": {},
          "performance": false,
          "warnHandler": undefined,
        },
        "directives": {},
        "emitsCache": WeakMap {},
        "mixins": [
          {
            "beforeCreate": [Function],
          },
        ],
        "optionsCache": WeakMap {},
        "propsCache": WeakMap {},
        "provides": {},
        "reload": [Function],
      },
      "asyncDep": null,
      "asyncResolved": false,
      "attrs": {},
      "bc": null,
      "bm": null,
      "bu": null,
      "bum": null,
      "c": null,
      "components": null,
      "ctx": {},
      "da": null,
      "data": {},
      "directives": null,
      "ec": null,
      "effect": ReactiveEffect {
        "active": true,
        "allowRecurse": true,
        "deps": [
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
        ],
        "fn": [Function],
        "onTrack": undefined,
        "onTrigger": undefined,
        "parent": undefined,
        "scheduler": [Function],
      },
      "emit": [Function],
      "emitsOptions": null,
      "emitted": null,
      "exposeProxy": null,
      "exposed": null,
      "inheritAttrs": undefined,
      "isDeactivated": false,
      "isMounted": true,
      "isUnmounted": false,
      "m": null,
      "next": null,
      "parent": null,
      "props": {},
      "propsDefaults": {},
      "propsOptions": [
        {},
        [],
      ],
      "provides": {},
      "proxy": {},
      "refs": {
        "VTU_COMPONENT": {
          "hasOwnProperty": [Function],
        },
      },
      "render": [Function],
      "renderCache": [],
      "root": [Circular],
      "rtc": null,
      "rtg": null,
      "scope": EffectScope {
        "active": true,
        "cleanups": [],
        "effects": [
          ReactiveEffect {
            "active": true,
            "allowRecurse": true,
            "deps": [
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
            ],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "parent": undefined,
            "scheduler": [Function],
          },
        ],
      },
      "setupContext": null,
      "setupState": {},
      "slots": {},
      "sp": null,
      "subTree": {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": null,
        "children": {
          "_ctx": [Circular],
          "default": [Function],
        },
        "component": {
          "a": null,
          "accessCache": {},
          "appContext": {
            "app": [Circular],
            "components": {
              "transition": {
                "name": "transition",
              },
              "transition-group": {
                "name": "transition-group",
              },
            },
            "config": {
              "compilerOptions": {},
              "errorHandler": undefined,
              "globalProperties": {},
              "isNativeTag": [Function],
              "optionMergeStrategies": {},
              "performance": false,
              "warnHandler": undefined,
            },
            "directives": {},
            "emitsCache": WeakMap {},
            "mixins": [
              {
                "beforeCreate": [Function],
              },
            ],
            "optionsCache": WeakMap {},
            "propsCache": WeakMap {},
            "provides": {},
            "reload": [Function],
          },
          "asyncDep": null,
          "asyncResolved": false,
          "attrs": {},
          "bc": null,
          "bm": null,
          "bu": null,
          "bum": null,
          "c": null,
          "components": {},
          "ctx": {
            "label": "Movies",
            "style": "default",
          },
          "da": null,
          "data": {},
          "devtoolsRawSetupState": {
            "part": ComputedRefImpl {
              "__v_isReadonly": true,
              "__v_isRef": true,
              "_cacheable": true,
              "_dirty": false,
              "_setter": [Function],
              "_value": {
                "component": [
                  "p-4 border border-gray-200 mb-4",
                ],
                "fieldsWrapper": [
                  "flex",
                ],
                "label": [
                  "text-sky-900 py-2 px-4 border border-gray-200 rounded",
                ],
              },
              "dep": Set {
                ReactiveEffect {
                  "active": true,
                  "allowRecurse": true,
                  "deps": [
                    [Circular],
                    Set {
                      [Circular],
                    },
                  ],
                  "fn": [Function],
                  "onTrack": undefined,
                  "onTrigger": undefined,
                  "parent": undefined,
                  "scheduler": [Function],
                },
              },
              "effect": ReactiveEffect {
                "active": true,
                "computed": [Circular],
                "deps": [
                  Set {
                    [Circular],
                  },
                ],
                "fn": [Function],
                "parent": undefined,
                "scheduler": [Function],
              },
            },
            "parts": {
              "component": {
                "style": {
                  "default": [
                    "p-4 border border-gray-200 mb-4",
                  ],
                },
              },
              "fieldsWrapper": {
                "style": {
                  "default": [
                    "flex",
                  ],
                },
              },
              "label": {
                "style": {
                  "default": [
                    "text-sky-900 py-2 px-4 border border-gray-200 rounded",
                  ],
                },
              },
            },
            "props": {
              "label": "Movies",
              "style": "default",
            },
          },
          "directives": null,
          "ec": null,
          "effect": ReactiveEffect {
            "active": true,
            "allowRecurse": true,
            "deps": [
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
            ],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "parent": undefined,
            "scheduler": [Function],
          },
          "emit": [Function],
          "emitsOptions": null,
          "emitted": null,
          "exposeProxy": {
            "hasOwnProperty": [Function],
          },
          "exposed": {
            "hasOwnProperty": [Function],
          },
          "inheritAttrs": undefined,
          "isDeactivated": false,
          "isMounted": true,
          "isUnmounted": false,
          "m": null,
          "next": null,
          "parent": [Circular],
          "props": {
            "label": "Movies",
            "style": "default",
          },
          "propsDefaults": {},
          "propsOptions": [
            {
              "label": {
                "0": false,
                "1": true,
                "required": true,
                "type": [Function],
              },
              "style": {
                "0": false,
                "1": true,
                "default": "default",
                "type": [Function],
              },
            },
            [
              "style",
            ],
          ],
          "provides": {},
          "proxy": {
            "label": "Movies",
            "style": "default",
          },
          "refs": {},
          "render": [Function],
          "renderCache": [],
          "root": [Circular],
          "rtc": null,
          "rtg": null,
          "scope": EffectScope {
            "active": true,
            "cleanups": [],
            "effects": [
              ReactiveEffect {
                "active": true,
                "computed": ComputedRefImpl {
                  "__v_isReadonly": true,
                  "__v_isRef": true,
                  "_cacheable": true,
                  "_dirty": false,
                  "_setter": [Function],
                  "_value": {
                    "component": [
                      "p-4 border border-gray-200 mb-4",
                    ],
                    "fieldsWrapper": [
                      "flex",
                    ],
                    "label": [
                      "text-sky-900 py-2 px-4 border border-gray-200 rounded",
                    ],
                  },
                  "dep": Set {
                    ReactiveEffect {
                      "active": true,
                      "allowRecurse": true,
                      "deps": [
                        [Circular],
                        Set {
                          [Circular],
                        },
                      ],
                      "fn": [Function],
                      "onTrack": undefined,
                      "onTrigger": undefined,
                      "parent": undefined,
                      "scheduler": [Function],
                    },
                  },
                  "effect": [Circular],
                },
                "deps": [
                  Set {
                    [Circular],
                  },
                ],
                "fn": [Function],
                "parent": undefined,
                "scheduler": [Function],
              },
              ReactiveEffect {
                "active": true,
                "allowRecurse": true,
                "deps": [
                  Set {
                    [Circular],
                  },
                  Set {
                    [Circular],
                  },
                ],
                "fn": [Function],
                "onTrack": undefined,
                "onTrigger": undefined,
                "parent": undefined,
                "scheduler": [Function],
              },
            ],
          },
          "setupContext": {
            "attrs": {},
            "emit": [Function],
            "expose": [Function],
            "slots": {
              "default": [Function],
            },
          },
          "setupState": {
            "part": {
              "component": [
                "p-4 border border-gray-200 mb-4",
              ],
              "fieldsWrapper": [
                "flex",
              ],
              "label": [
                "text-sky-900 py-2 px-4 border border-gray-200 rounded",
              ],
            },
            "parts": {
              "component": {
                "style": {
                  "default": [
                    "p-4 border border-gray-200 mb-4",
                  ],
                },
              },
              "fieldsWrapper": {
                "style": {
                  "default": [
                    "flex",
                  ],
                },
              },
              "label": {
                "style": {
                  "default": [
                    "text-sky-900 py-2 px-4 border border-gray-200 rounded",
                  ],
                },
              },
            },
            "props": {
              "label": "Movies",
              "style": "default",
            },
          },
          "slots": {
            "default": [Function],
          },
          "sp": null,
          "subTree": {
            "__v_isVNode": true,
            "__v_skip": true,
            "anchor": null,
            "appContext": null,
            "children": [
              {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": "Movies",
                "component": null,
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": <legend
                  class="text-sky-900 py-2 px-4 border border-gray-200 rounded"
                  data-part="label"
                >
                  Movies
                </legend>,
                "key": null,
                "patchFlag": 3,
                "props": {
                  "class": "text-sky-900 py-2 px-4 border border-gray-200 rounded",
                  "data-part": "label",
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 9,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "legend",
              },
              {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": [
                  {
                    "__v_isVNode": true,
                    "__v_skip": true,
                    "anchor": ,
                    "appContext": null,
                    "children": [
                      {
                        "__v_isVNode": true,
                        "__v_skip": true,
                        "anchor": null,
                        "appContext": null,
                        "children": "Hello world",
                        "component": null,
                        "dirs": null,
                        "dynamicChildren": null,
                        "dynamicProps": null,
                        "el": Hello world,
                        "key": null,
                        "patchFlag": 0,
                        "props": null,
                        "ref": null,
                        "scopeId": null,
                        "shapeFlag": 8,
                        "slotScopeIds": null,
                        "ssContent": null,
                        "ssFallback": null,
                        "staticCount": 0,
                        "suspense": null,
                        "target": null,
                        "targetAnchor": null,
                        "transition": null,
                        "type": Symbol(Text),
                      },
                    ],
                    "component": null,
                    "dirs": null,
                    "dynamicChildren": null,
                    "dynamicProps": null,
                    "el": ,
                    "key": "_default",
                    "patchFlag": -2,
                    "props": {
                      "key": "_default",
                    },
                    "ref": null,
                    "scopeId": null,
                    "shapeFlag": 16,
                    "slotScopeIds": null,
                    "ssContent": null,
                    "ssFallback": null,
                    "staticCount": 0,
                    "suspense": null,
                    "target": null,
                    "targetAnchor": null,
                    "transition": null,
                    "type": Symbol(Fragment),
                  },
                ],
                "component": null,
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": <div
                  class="flex"
                  data-part="fieldsWrapper"
                >
                  
                  Hello world
                  
                </div>,
                "key": null,
                "patchFlag": 2,
                "props": {
                  "class": "flex",
                  "data-part": "fieldsWrapper",
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 17,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "div",
              },
            ],
            "component": null,
            "dirs": null,
            "dynamicChildren": [
              {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": "Movies",
                "component": null,
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": <legend
                  class="text-sky-900 py-2 px-4 border border-gray-200 rounded"
                  data-part="label"
                >
                  Movies
                </legend>,
                "key": null,
                "patchFlag": 3,
                "props": {
                  "class": "text-sky-900 py-2 px-4 border border-gray-200 rounded",
                  "data-part": "label",
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 9,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "legend",
              },
              {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": ,
                "appContext": null,
                "children": [
                  {
                    "__v_isVNode": true,
                    "__v_skip": true,
                    "anchor": null,
                    "appContext": null,
                    "children": "Hello world",
                    "component": null,
                    "dirs": null,
                    "dynamicChildren": null,
                    "dynamicProps": null,
                    "el": Hello world,
                    "key": null,
                    "patchFlag": 0,
                    "props": null,
                    "ref": null,
                    "scopeId": null,
                    "shapeFlag": 8,
                    "slotScopeIds": null,
                    "ssContent": null,
                    "ssFallback": null,
                    "staticCount": 0,
                    "suspense": null,
                    "target": null,
                    "targetAnchor": null,
                    "transition": null,
                    "type": Symbol(Text),
                  },
                ],
                "component": null,
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": ,
                "key": "_default",
                "patchFlag": -2,
                "props": {
                  "key": "_default",
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 16,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": Symbol(Fragment),
              },
              {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": [
                  {
                    "__v_isVNode": true,
                    "__v_skip": true,
                    "anchor": ,
                    "appContext": null,
                    "children": [
                      {
                        "__v_isVNode": true,
                        "__v_skip": true,
                        "anchor": null,
                        "appContext": null,
                        "children": "Hello world",
                        "component": null,
                        "dirs": null,
                        "dynamicChildren": null,
                        "dynamicProps": null,
                        "el": Hello world,
                        "key": null,
                        "patchFlag": 0,
                        "props": null,
                        "ref": null,
                        "scopeId": null,
                        "shapeFlag": 8,
                        "slotScopeIds": null,
                        "ssContent": null,
                        "ssFallback": null,
                        "staticCount": 0,
                        "suspense": null,
                        "target": null,
                        "targetAnchor": null,
                        "transition": null,
                        "type": Symbol(Text),
                      },
                    ],
                    "component": null,
                    "dirs": null,
                    "dynamicChildren": null,
                    "dynamicProps": null,
                    "el": ,
                    "key": "_default",
                    "patchFlag": -2,
                    "props": {
                      "key": "_default",
                    },
                    "ref": null,
                    "scopeId": null,
                    "shapeFlag": 16,
                    "slotScopeIds": null,
                    "ssContent": null,
                    "ssFallback": null,
                    "staticCount": 0,
                    "suspense": null,
                    "target": null,
                    "targetAnchor": null,
                    "transition": null,
                    "type": Symbol(Fragment),
                  },
                ],
                "component": null,
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": <div
                  class="flex"
                  data-part="fieldsWrapper"
                >
                  
                  Hello world
                  
                </div>,
                "key": null,
                "patchFlag": 2,
                "props": {
                  "class": "flex",
                  "data-part": "fieldsWrapper",
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 17,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "div",
              },
            ],
            "dynamicProps": [
              "data-component",
            ],
            "el": <fieldset
              class="p-4 border border-gray-200 mb-4"
              data-component="BonesFormFieldSet"
              data-part="component"
            >
              <legend
                class="text-sky-900 py-2 px-4 border border-gray-200 rounded"
                data-part="label"
              >
                Movies
              </legend>
              <div
                class="flex"
                data-part="fieldsWrapper"
              >
                
                Hello world
                
              </div>
            </fieldset>,
            "key": null,
            "patchFlag": 10,
            "props": {
              "class": "p-4 border border-gray-200 mb-4",
              "data-component": "BonesFormFieldSet",
              "data-part": "component",
            },
            "ref": null,
            "scopeId": null,
            "shapeFlag": 17,
            "slotScopeIds": null,
            "ssContent": null,
            "ssFallback": null,
            "staticCount": 0,
            "suspense": null,
            "target": null,
            "targetAnchor": null,
            "transition": null,
            "type": "fieldset",
          },
          "suspense": null,
          "suspenseId": 0,
          "type": {
            "__file": "/Users/liamhall/Documents/web-development/applications/barebones/src/components/FormFieldSet/FormFieldSet.vue",
            "__hmrId": "5b076dcf",
            "components": {},
            "name": "BonesFormFieldSet",
            "props": {
              "label": {
                "0": false,
                "1": true,
                "required": true,
                "type": [Function],
              },
              "style": {
                "0": false,
                "1": true,
                "default": "default",
                "type": [Function],
              },
            },
            "render": [Function],
            "setup": [Function],
          },
          "u": null,
          "uid": 7,
          "um": null,
          "update": [Function],
          "vnode": [Circular],
          "withProxy": null,
        },
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": <fieldset
          class="p-4 border border-gray-200 mb-4"
          data-component="BonesFormFieldSet"
          data-part="component"
        >
          <legend
            class="text-sky-900 py-2 px-4 border border-gray-200 rounded"
            data-part="label"
          >
            Movies
          </legend>
          <div
            class="flex"
            data-part="fieldsWrapper"
          >
            
            Hello world
            
          </div>
        </fieldset>,
        "key": null,
        "patchFlag": 0,
        "props": {
          "label": "Movies",
          "ref": "VTU_COMPONENT",
        },
        "ref": {
          "f": false,
          "i": [Circular],
          "k": undefined,
          "r": "VTU_COMPONENT",
        },
        "scopeId": null,
        "shapeFlag": 36,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "transition": null,
        "type": {
          "__file": "/Users/liamhall/Documents/web-development/applications/barebones/src/components/FormFieldSet/FormFieldSet.vue",
          "__hmrId": "5b076dcf",
          "components": {},
          "name": "BonesFormFieldSet",
          "props": {
            "label": {
              "0": false,
              "1": true,
              "required": true,
              "type": [Function],
            },
            "style": {
              "0": false,
              "1": true,
              "default": "default",
              "type": [Function],
            },
          },
          "render": [Function],
          "setup": [Function],
        },
      },
      "suspense": null,
      "suspenseId": 0,
      "type": {
        "name": "VTU_ROOT",
        "render": [Function],
      },
      "u": null,
      "uid": 6,
      "um": null,
      "update": [Function],
      "vnode": {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": {
          "app": [Circular],
          "components": {
            "transition": {
              "name": "transition",
            },
            "transition-group": {
              "name": "transition-group",
            },
          },
          "config": {
            "compilerOptions": {},
            "errorHandler": undefined,
            "globalProperties": {},
            "isNativeTag": [Function],
            "optionMergeStrategies": {},
            "performance": false,
            "warnHandler": undefined,
          },
          "directives": {},
          "emitsCache": WeakMap {},
          "mixins": [
            {
              "beforeCreate": [Function],
            },
          ],
          "optionsCache": WeakMap {},
          "propsCache": WeakMap {},
          "provides": {},
          "reload": [Function],
        },
        "children": null,
        "component": [Circular],
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": <fieldset
          class="p-4 border border-gray-200 mb-4"
          data-component="BonesFormFieldSet"
          data-part="component"
        >
          <legend
            class="text-sky-900 py-2 px-4 border border-gray-200 rounded"
            data-part="label"
          >
            Movies
          </legend>
          <div
            class="flex"
            data-part="fieldsWrapper"
          >
            
            Hello world
            
          </div>
        </fieldset>,
        "key": null,
        "patchFlag": 0,
        "props": null,
        "ref": null,
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "transition": null,
        "type": {
          "name": "VTU_ROOT",
          "render": [Function],
        },
      },
      "withProxy": null,
    },
    "_props": null,
    "_uid": 3,
    "component": [Function],
    "config": {
      "compilerOptions": {},
      "errorHandler": undefined,
      "globalProperties": {},
      "isNativeTag": [Function],
      "optionMergeStrategies": {},
      "performance": false,
      "warnHandler": undefined,
    },
    "directive": [Function],
    "mixin": [Function],
    "mount": [Function],
    "provide": [Function],
    "unmount": [Function],
    "use": [Function],
    "version": "3.2.33",
  },
  "__setProps": [Function],
  "componentVM": {
    "label": "Movies",
    "style": "default",
  },
  "isDisabled": [Function],
  "rootVM": {},
  "wrapperElement": <fieldset
    class="p-4 border border-gray-200 mb-4"
    data-component="BonesFormFieldSet"
    data-part="component"
  >
    <legend
      class="text-sky-900 py-2 px-4 border border-gray-200 rounded"
      data-part="label"
    >
      Movies
    </legend>
    <div
      class="flex"
      data-part="fieldsWrapper"
    >
      
      Hello world
      
    </div>
  </fieldset>,
}
`;
